api/grpcserver/mesh_service.go

```diff
@@ -252,15 +252,19 @@ func castTransaction(t *types.Transaction) *pb.Transaction {
 	return tx
 }
 
-func convertActivation(a *types.ActivationTx) *pb.Activation {
+func convertActivation(a *types.ActivationTx, previous []types.ATXID) *pb.Activation {
+	if len(previous) == 0 {
+		previous = []types.ATXID{types.EmptyATXID}
+	}
 	return &pb.Activation{
 		Id:        &pb.ActivationId{Id: a.ID().Bytes()},
 		Layer:     &pb.LayerNumber{Number: a.PublishEpoch.Uint32()},
 		SmesherId: &pb.SmesherId{Id: a.SmesherID.Bytes()},
 		Coinbase:  &pb.AccountId{Address: a.Coinbase.String()},
-		PrevAtx:   &pb.ActivationId{Id: a.PrevATXID.Bytes()},
-		NumUnits:  uint32(a.NumUnits),
-		Sequence:  a.Sequence,
+		// TODO: should we update the API to support multiple previous ATXs?
+		PrevAtx:  &pb.ActivationId{Id: previous[0].Bytes()},
```

Comment:
Maybe, for now I think its OK to return the previous ATX of the identity that created/signed the ATX.

Reply:
I think it doesn't make sense to invest in it as we will retire the `v1` API soon anyway.

Reply:
`previous[0]` is the previous ATX of the identity that created the ATX or isn't it?

Reply:
There's no such guarantee.